server/src/main/java/io/deephaven/server/session/SessionState.java

```diff
@@ -1141,6 +1141,7 @@ protected synchronized void destroy() {
             if (!(caughtException instanceof StatusRuntimeException)) {
                 caughtException = null;
             }
+            queryPerformanceRecorder = null;
```

Comment:
What is the goal here? I assume we're not keeping hard refs to the `ExportObject` once it's destroyed most of the time, anyway.

Reply:
At this time, we do keep hard refs to ExportObjects within a session so that we can tell if an unrecognized export ID is either stale RELEASED ("sorry, you raced but you just missed it, that was already released") or UNKNOWN ("hmm, never seen that before, you probably raced it and won, so we'll wait until it appears").

The linked issue mentions a strategy to handle this, with a RowSet for known-released objects, so that we no longer need to keep the ExportObject itself around.